### PR TITLE
CLOUDP-328313: Quickfix for legacy custom methods

### PR DESCRIPTION
**Pull request by sphterry**
Debug for legacy custom methods case 

_Jira ticket:_ [CLOUDP-328313](https://jira.mongodb.org/browse/CLOUDP-328313)

### CODE REVIEW

File: tools/spectral/ipa/rulesets/functions/utils/operationIdGeneration.js [hunk 1]
```diff
@@ -24,14 +24,16 @@ export function generateOperationID(method, path) {
   // legacy custom method - use end of path as custom method name
```
Comment (wtrocki) [inline tools/spectral/ipa/rulesets/functions/utils/operationIdGeneration.js:24]: [nit] can be separate method for handling legacy operations right?

File: tools/spectral/ipa/rulesets/functions/utils/operationIdGeneration.js [hunk 2]
```diff
@@ -24,6 +24,7 @@ export function generateOperationID(method, path) {
...
   if (!method) {
     method = nouns.pop();
+    resourceIdentifier = resourceIdentifier.slice(0, resourceIdentifier.lastIndexOf('/'));
```
[the remark this answers is not in the document]
Comment (wtrocki) [reply tools/spectral/ipa/rulesets/functions/utils/operationIdGeneration.js:27]: [nit] Do we need test for empty operationIds/resourceIdentifier not in the shape we expected.

File: tools/spectral/ipa/rulesets/functions/utils/operationIdGeneration.js (hunk 1 above)
Comment (sphterry) [reply tools/spectral/ipa/rulesets/functions/utils/operationIdGeneration.js:24]: Yes, I could probably make a method that contains the extra logic and calls on this function. Will be something to consider when I'm implementing the verbosity override - might refactor for use in generating a shorter recommended opId
